backend/db.js

```diff
@@ -0,0 +1,60 @@
+import { DataSource } from "apollo-datasource";
+import crypto from "crypto";
+
+export class User {
+  constructor(data) {
+    Object.assign(this, data);
+  }
+}
+export class Post {
+  constructor(data) {
+    this.id = crypto.randomBytes(16).toString("hex");
```

Comment:
Its rare but possible to have duplicate postids, also very expensive call.
how about removing and adding in addNewPost(postInput)?